packages/fgtclb/academic-jobs/Classes/Property/TypeConverter/ImageUploadConverter.php

```diff
@@ -14,15 +16,18 @@
 use TYPO3\CMS\Core\Resource\ResourceFactory;
 use TYPO3\CMS\Core\Resource\Security\FileNameValidator;
 use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\StringUtility;
 use TYPO3\CMS\Extbase\Domain\Model\FileReference as ExtbaseFileReference;
 use TYPO3\CMS\Extbase\Error\Error;
 use TYPO3\CMS\Extbase\Property\Exception\TypeConverterException;
 use TYPO3\CMS\Extbase\Property\PropertyMappingConfigurationInterface;
 use TYPO3\CMS\Extbase\Property\TypeConverter\AbstractTypeConverter;
 use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
 
-final class JobAvatarImageUploadConverter extends AbstractTypeConverter
+final class ImageUploadConverter extends AbstractTypeConverter implements LoggerAwareInterface
 {
+    use LoggerAwareTrait;
```

Comment:
Why switching from constructor DI (ensured retrieved logger) to the "eventually available" using the trait ? (I don't like the trait) Special when constructor is safe to be used for DI.